darts/models/forecasting/nlinear.py

```diff
@@ -0,0 +1,246 @@
+"""
+N-Linear
+------
+https://github.com/cure-lab/LTSF-Linear
+
+Copyright 2022 DLinear Authors. All rights reserved.
+
+
+"""
+
+import torch
+import torch.nn as nn
+
+from darts.models.forecasting.pl_forecasting_module import PLPastCovariatesModule
+from darts.models.forecasting.torch_forecasting_model import PastCovariatesTorchModel
+
+
+class _NLinearModule(PLPastCovariatesModule):
+    """
+    NLinear
+    """
+
+    def __init__(
+        self,
+        input_dim,
+        output_dim,
+        nr_params,
+        **kwargs,
+    ):
+        """PyTorch module implementing the N-HiTS architecture.
+
+        Parameters
+        ----------
+        input_dim
+            The number of input components (target + optional covariate)
+        output_dim
+            Number of output components in the target
+        nr_params
+            The number of parameters of the likelihood (or 1 if no likelihood is used).
+        **kwargs
+            all parameters required for :class:`darts.model.forecasting_models.PLForecastingModule` base class.
+
+        Inputs
+        ------
+        x of shape `(batch_size, input_chunk_length)`
+            Tensor containing the input sequence.
+
+        Outputs
+        -------
+        y of shape `(batch_size, output_chunk_length, target_size/output_dim, nr_params)`
+            Tensor containing the output of the NBEATS module.
+
+        """
+        super().__init__(**kwargs)
+        self.input_dim = input_dim
+        self.output_dim = output_dim
+        self.nr_params = nr_params
+        self.seq_len = self.input_chunk_length
+        self.pred_len = self.output_chunk_length
+
+        self.Linear = nn.Linear(self.seq_len, self.pred_len * self.nr_params)
+
+    def forward(self, x_in: tuple):
+        x, _ = x_in
+        # x: [Batch, Input length, Channel]
+
+        seq_last = x[:, -1:, :].detach()
+        x = x - seq_last
+        x = self.Linear(x.permute(0, 2, 1)).permute(0, 2, 1)
+        x = x + seq_last
+
+        x = x.view(
```

Comment:
@hrzn, so I've been working on this for a while. `x` is of shape `batch_size, output_chunk_length * nr_params, input_dim`. `view` cannot split up the dims into `batch_size, output_chunk_length, input_dim, nr_params`. I can do a permute right before this, but the accuracy goes way down. I've also tried  `reshape` and a few other things but the predictions get way worse. 

Do you have any thoughts?

Reply:
Thanks for the update @gdevos010. I'll have a look and make a suggestion.